daffodil-lib/src/main/scala/org/apache/daffodil/lib/xml/DaffodilXMLLoader.scala

```diff
@@ -228,19 +228,19 @@ class DFDLCatalogResolver private ()
      *    <xs:schema...
      *    ... xsi:schemaLocation="urn:some:namespace /some/path.xsd"
      *  }}}
-     * Xerces takes that schemaLocation URI and absolutizes it to {{{ file:/some/path.xsd }}}
+     * Xerces takes that schemaLocation URI and absolutizes it to {{{ URISCHEME:/some/path.xsd }}}
      * and passes that to our resolveEntity and in turn resolveCommon, which while it's able
-     * to find the namespace, fails to set the resolvedUri since the file:/some/path.xsd will
+     * to find the namespace, fails to set the resolvedUri since the URISCHEME:/some/path.xsd will
      * never match anything resolved from our catalog since that'd return something like
-     * {{{ file:/some/absolute/path/to/some/path.xsd }}}
+     * {{{ URISCHEME:/some/absolute/path/to/some/path.xsd }}}
      *
      * This is a workaround to that bug where we convert systemId to a URI and check if the
      * path (from URI.getPath) matches the end of resolvedUri. Note: This can ignore absolute
      * URIs passed in for schemaLocation, but those are edge cases where the user expects
      * the namespace to match a different file (i.e what they provide in the schemalocation)
      * than what we find in the catalog.
      */
-    lazy val systemIdPath = if (systemIdUri != null && systemIdUri.getScheme == "file") {
+    lazy val systemIdPath = if (systemIdUri != null && systemIdUri.getScheme != null) {
```

Comment:
I think this changes behavior if the systemIdUri schema is "jar". Previously "jar" schemes would fall into the false block but now they fall into the true block and jarURI.getPath will return null. It's hard to follow what all this code is doing, but feels like that might break things?

Reply:
So I tested this out by adding an xsi:schemaLocation to DFDLGeneralFormat and then did some prinlns, I got the below which means jarUri.getPath won't return null. I also think this isn't a typical jarURI so maybe it's fine?

sysId: jar:/org/apache/daffodil/xsd/XMLSchema_for_DFDL.xsd
.getScheme: jar
.getPath: /org/apache/daffodil/xsd/XMLSchema_for_DFDL.xsd

Reply:
Ah right, because Xerces doesn't actually create a real jar URI, it just takes the path and prepends a scheme for no reason. So we're just stripping off that incorrect scheme to get the original path. This makes sense then.